### PR TITLE
Allow export default interface. Related to issue 3792.

### LINKED ISSUE #3792: Cannot 'export default' abstract, ambient class or interface
```
export default abstract class B {

}
```

By the way, the same with

```
export default declare class B {

}
```

Seems to be a parser bug

### PR DESCRIPTION
**Pull request by erikmcc**
Fixes #3792.

### CODE REVIEW

**Comment by erikmcc** [comment]
Has the CLA changed? I am pretty sure I signed it before...

File: tests/baselines/reference/exportDefaultInterface.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+//// [tests/cases/compiler/exportDefaultInterface.ts] ////
+
+//// [a.ts]
+export default interface A {}
+
+//// [b.ts]
+import A from './a'
```
Comment (mhegazy) [inline tests/baselines/reference/exportDefaultInterface.js]: Please add a test where `A` is actually used.

**Comment by mhegazy** [comment]
@andy-ms can you please review this change.

**Comment by mhegazy** [comment]
I pushed tests and fixed resolveName in https://github.com/Microsoft/TypeScript/pull/16040

**Comment by mhegazy** [comment]
closing in favor of https://github.com/Microsoft/TypeScript/pull/16040

**Pull request closed without merging**